EIPS/eip-draft-ens-text-data.md

```diff
@@ -0,0 +1,72 @@
+## Preamble
+
+    EIP: <to be assigned>
+    Title: Storage of text records in ENS 
+    Author: Richard Moore <me@ricmoo.com>
+    Type: Standard Track
+    Category ERC
+    Status: Draft
+    Created: 2017-05-17
+
+
+## Abstract
+This EIP defines a resolver profile for ENS that permits the lookup of arbitrary key-value
+text data. This allows ENS name holders to associate e-mail addresses, URLs and other
+informational data with a ENS name.
+
+
+## Motivation
+There is often a desire for human-readable metadata to be associated with otherwise
+machine-driven data; used for debugging, maintenance, reporting and general information.
+
+In this EIP we define a simple resolver profile for ENS that permits ENS names to
+associate arbitrary key-value text.
+
+
+## Specification
+
+### Resolver Profile
+A new resolver interface is defined, consisting of the following method:
+
+    function text(bytes32 node, string key) constant returns (string text);
+
+The interface ID of this interface is 0x59d1d43c.
+
+The `text` data may be any arbitrary UTF-8 string. If the key is not present, the empty string
+must be returned.
+
+
+### Initial Recommended Keys
+
+Keys must be made up of lowercase letters, numbers and the hyphen (-). Vendor specific
+services must be prefixed with `vnd.`.
+
+- **email** - an e-mail address
+- **url** - a URL
+- **avatar** - a URL to an image used as an avatar or logo
+- **description** - A description of the name
+- **notice** - A notice regarding this name; 
+- **keywords** - A list of comma-separated keywords, ordered by most significant first; clients that interpresent this field may choose a threshold beyond which to ignore
+- **vnd.twitter** - a twitter username (should it be prefixed with an @?)
+- **vnd.github** - a GitHub username
+
```

Comment:
Might be worth adding:

* Name
* Address
* Geo coordinates
* Phone number

This would enable customers to quickly search which businesses nearby accept Ether.

Reply:
but would also be a big privacy issue ..-)

Reply:
Sorry, I didn’t notice this comment earlier. :)

Yes, I think those make sense...

Regarding privacy, it needs to be properly conveyed in tools that this information is public. It is the same as a person or company including this data on their website.